main.go

```diff
@@ -52,7 +53,45 @@ func main() {
 		log.Fatal(errors.AddContext(err, "failed to build the api"))
 	}
 
-	logger.Printf("Starting Pinner service.\nGitRevision: %v (built %v)\n", build.GitRevision, build.BuildTime)
+	logger.Print("Starting Pinner service")
+	logger.Printf("GitRevision: %v (built %v)", build.GitRevision, build.BuildTime)
 	err = server.ListenAndServe(4000)
 	log.Fatal(errors.Compose(err, scanner.Close()))
 }
+
+// newLogger creates a new logger that can write to disk.
+//
+// The function also returns a closer function that should be called when we
+// stop using the logger, typically deferred in main.
+func newLogger(level, logfile string) (logger *logrus.Logger, closerFn func(), err error) {
+	logger = logrus.New()
+	// Parse and set log level.
+	logLevel, err := logrus.ParseLevel(level)
+	if err != nil {
+		return nil, nil, errors.AddContext(err, "invalid log level: "+level)
+	}
+	logger.SetLevel(logLevel)
+	// Open and start writing to the log file, unless we have an empty string,
+	// which signifies "don't log to disk".
+	if logfile != "" {
```

Comment:
What happens when you specify a path to a directory that doesn't exist? It'll error out right?